MathLib/LinAlg/MatrixVectorTraits.cpp

```diff
@@ -84,7 +92,32 @@ std::unique_ptr<PETScMatrix>
 MatrixVectorTraits<PETScMatrix>::
 newInstance(MatrixSpecifications const& spec)
 {
-    return std::unique_ptr<PETScMatrix>(new PETScMatrix(spec.nrows)); // TODO sparsity pattern
+    auto const nrows = spec.dof_table ? spec.dof_table->dofSizeLocal() : spec.nrows;
+    auto const ncols = spec.dof_table ? nrows : spec.ncols;
+
+    // TODO I guess it is not hard to make AssemblerLib::computeSparsityPattern()
+    //      work also for NodePartitionedMesh'es.
+    assert(((!spec.sparsity_pattern) || spec.sparsity_pattern->size() == 0u) &&
```

Comment:
Currently, if I remember correctly, we don't precompute a sparsity pattern in the PETSc build. Rather we get it from the node partitioned mesh in this method. I think this assertion should stay as long as we don't precompute precise sparsity patterns for PETSc.